packages/amplify-graphql-auth-transformer/src/utils/iam.ts

```diff
@@ -37,7 +38,7 @@ export const createPolicyDocumentForManagedPolicy = (resources: Set<string>) =>
     if (fieldName !== 'null') {
       policyDocumentResources.push(
         cdk.Fn.sub('arn:aws:appsync:${AWS::Region}:${AWS::AccountId}:apis/${apiId}/types/${typeName}/fields/${fieldName}', {
-          apiId: cdk.Fn.getAtt('GraphQLAPI', 'ApiId').toString(),
+          apiId: context.api.apiId,
```

Comment:
Where do we set the `context.api.apiId`? Want to make sure it is set in transformer context for CLI and CDK flows.

Reply:
> Where do we set the `context.api.apiId`? Want to make sure it is set in transformer context for CLI and CDK flows.

It's set in the graphqlapi class itself - so should be correct here https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-graphql-transformer-core/src/graphql-api.ts#L219

I've also confirmed via a `setup-dev` that it's passing through as expected.

```json
"AuthRolePolicy01921FC820": {
    "Type": "AWS::IAM::ManagedPolicy",
    "Properties": {
        "PolicyDocument": {
            "Statement": [
                {
                    "Action": "appsync:GraphQL",
                    "Effect": "Allow",
                    "Resource": [
                        {
                            "Fn::Sub": [
                                "arn:aws:appsync:${AWS::Region}:${AWS::AccountId}:apis/${apiId}/types/${typeName}/*",
                                {
                                    "apiId": {
                                        "Fn::GetAtt": [
                                            "GraphQLAPI",
                                            "ApiId"
                                        ]
                                    },
                                    "typeName": "Todo"
                                }
                            ]
                        },
                        {
                            "Fn::Sub": [
                                "arn:aws:appsync:${AWS::Region}:${AWS::AccountId}:apis/${apiId}/types/${typeName}/fields/${fieldName}",
                                {
                                    "apiId": {
                                        "Fn::GetAtt": [
                                            "GraphQLAPI",
                                            "ApiId"
                                        ]
                                    },
                                    "typeName": "Query",
                                    "fieldName": "getTodo"
                                }
                            ]
                        },
                        {
                            "Fn::Sub": [
                                "arn:aws:appsync:${AWS::Region}:${AWS::AccountId}:apis/${apiId}/types/${typeName}/fields/${fieldName}",
                                {
                                    "apiId": {
                                        "Fn::GetAtt": [
                                            "GraphQLAPI",
                                            "ApiId"
                                        ]
                                    },
                                    "typeName": "Query",
                                    "fieldName": "listTodos"
                                }
                            ]
                        },
                        {
                            "Fn::Sub": [
                                "arn:aws:appsync:${AWS::Region}:${AWS::AccountId}:apis/${apiId}/types/${typeName}/fields/${fieldName}",
                                {
                                    "apiId": {
                                        "Fn::GetAtt": [
                                            "GraphQLAPI",
                                            "ApiId"
                                        ]
                                    },
                                    "typeName": "Mutation",
                                    "fieldName": "createTodo"
                                }
                            ]
                        },
                        {
                            "Fn::Sub": [
                                "arn:aws:appsync:${AWS::Region}:${AWS::AccountId}:apis/${apiId}/types/${typeName}/fields/${fieldName}",
                                {
                                    "apiId": {
                                        "Fn::GetAtt": [
                                            "GraphQLAPI",
                                            "ApiId"
                                        ]
                                    },
                                    "typeName": "Mutation",
                                    "fieldName": "updateTodo"
                                }
                            ]
                        },
                        {
                            "Fn::Sub": [
                                "arn:aws:appsync:${AWS::Region}:${AWS::AccountId}:apis/${apiId}/types/${typeName}/fields/${fieldName}",
                                {
                                    "apiId": {
                                        "Fn::GetAtt": [
                                            "GraphQLAPI",
                                            "ApiId"
                                        ]
                                    },
                                    "typeName": "Mutation",
                                    "fieldName": "deleteTodo"
                                }
                            ]
                        },
                        {
                            "Fn::Sub": [
                                "arn:aws:appsync:${AWS::Region}:${AWS::AccountId}:apis/${apiId}/types/${typeName}/fields/${fieldName}",
                                {
                                    "apiId": {
                                        "Fn::GetAtt": [
                                            "GraphQLAPI",
                                            "ApiId"
                                        ]
                                    },
                                    "typeName": "Subscription",
                                    "fieldName": "onCreateTodo"
                                }
                            ]
                        },
                        {
                            "Fn::Sub": [
                                "arn:aws:appsync:${AWS::Region}:${AWS::AccountId}:apis/${apiId}/types/${typeName}/fields/${fieldName}",
                                {
                                    "apiId": {
                                        "Fn::GetAtt": [
                                            "GraphQLAPI",
                                            "ApiId"
                                        ]
                                    },
                                    "typeName": "Subscription",
                                    "fieldName": "onUpdateTodo"
                                }
                            ]
                        },
                        {
                            "Fn::Sub": [
                                "arn:aws:appsync:${AWS::Region}:${AWS::AccountId}:apis/${apiId}/types/${typeName}/fields/${fieldName}",
                                {
                                    "apiId": {
                                        "Fn::GetAtt": [
                                            "GraphQLAPI",
                                            "ApiId"
                                        ]
                                    },
                                    "typeName": "Subscription",
                                    "fieldName": "onDeleteTodo"
                                }
                            ]
                        }
                    ]
                }
            ],
            "Version": "2012-10-17"
        },
        "Description": "",
        "Path": "/",
        "Roles": [
            {
                "Ref": "authRoleName"
            }
        ]
    }
}
```